### DISCUSSION TITLE
multiple devcontainer config files in sub directories

### DISCUSSION THREAD

**Discussion by hellt** [loft-sh/devpod#1447] [Q&A] [answered]
Hi all,
first of all, thanks for a great product/project.

I wonder, is there a nice way to give a user a choice of which dev container file to use when they build a workspace from a folder/repo?

We keep several devcontainer files under nested dirs, in the following layout:

```
.devcontainer
├── docker-in-docker
│   └── devcontainer.json
└── docker-outside-of-docker
    └── devcontainer.json
```

I was hoping that this setting would work
<img width="644" alt="image" src="https://github.com/user-attachments/assets/87f0977b-ca06-4b3e-98e4-8efbafabae72">
but it does not probably look into the nested directories.

Mayb if it is a lot of work to have a dialog to open up a certain devcontainer file, then maybe we could have an option added to the URL interceptor so that we can encode which path to use for the devcontainer like this:

```
https://devpod.sh/open#https://github.com/loft-sh/devpod?devcontainer-path=.devcontainer/somedir/devcontainer.json
```

(codespaces does it `https://codespaces.new/srl-labs/srl-snmp-framework-lab?quickstart=1&devcontainer_path=.devcontainer%2Fdocker-in-docker%2Fdevcontainer.json`)

**Comment by pascalbreuninger** [comment]
Hey @hellt, we've fixed the multi devcontainer detection and it'll be working again with the next release. It's opt-in because we need to clone the repository before we can start looking for devcontainer.jsons and this might take a couple of seconds.

Specifying the devcontainer.json through `open` also makes a lot of sense, I'll create a ticket for it

**Reply by hellt** [reply]
oh great! Thanks @pascalbreuninger ! Looking forward for the deeplink for the devcontainer selection. This will streamline it for us so much!

**Reply by hellt** [reply, marked as answer]
I checked the 0.6.10 and now when several devcontainer json files are present, the popup at the create workspace stage offers to choose one. Works perfectly, thanks!
